MapboxCoreNavigation/NavigationEventsManager.swift

```diff
@@ -2,33 +2,36 @@ import Foundation
 import MapboxMobileEvents
 import MapboxDirections
 
-/**
- The `EventsManager` is responsible for being the liaison between MapboxCoreNavigation and the telemetry framework.
- 
- `SessionState` is a struct that stores all memoized statistics that we later send to the telemetry engine.
+ /**
+ The `EventsManagerDataSource` protocol declares values required for recording route following events.
  */
-
 @objc public protocol EventsManagerDataSource: class {
     var routeProgress: RouteProgress { get }
     var location: CLLocation? { get }
     var desiredAccuracy: CLLocationAccuracy { get }
     var locationProvider: NavigationLocationManager.Type { get }
 }
 
-@objc(MBEventsManager)
-open class EventsManager: NSObject {
-    
-    @objc public var manager: MMEEventsManager = .shared()
-    
+public typealias EventsManager = NavigationEventsManager
+
+/**
+ The `NavigationEventsManager` is responsible for being the liaison between MapboxCoreNavigation and the Mapbox telemetry framework.
+ */
+@objc(MBNavigationEventsManager)
+open class NavigationEventsManager: NSObject {
+
     var sessionState: SessionState!
```

Comment:
Noting here that this class handles `sessionState` inconsistently internally. Paging @JThramer , I think we should address this as a part of this PR.

Reply:
👀

Reply:
I’m referring to the inconsistent use of ? when accessing sessionState. Now that dataSource is optional, we might need to do more to ensure sessionState is always non-nil, it unwrapped safely